runtime/Java/src/org/antlr/v4/runtime/atn/LexerATNSimulator.java

```diff
@@ -735,8 +734,13 @@ public void consume(CharStream input) {
 		if ( curChar=='\n' ) {
 			line++;
 			charPositionInLine=0;
-		}
-		else {
+		} else if ( curChar == '\r' ) {
+			int nextChar = input.LA(2);
+			if ( nextChar != '\n') {
+				line++;
+				charPositionInLine=0;
+			}
+		} else {
```

Comment:
Maybe it's better to change to the following?

```suggestion
		} else if ( curChar == '\r' ) {
			if (input.LA(2) == '\n') {
			    input.consume();
			}
			line++;
		    charPositionInLine=0;
		} else {
```

It looks more optimal since it does not require reading `\n` twice.